### PR TITLE
Feature/31510 Search for duplicate UPN within school

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
- [x] Lookup upn within school during pupil validation process for both single and add multiple pupil submission
- [x] Migration change to allow duplicate UPNs across schools, but not within a school

### CODE REVIEW

File: admin/data/sql/migrations/20190402121931.do.add-unique-undex-pupil-table-upn-and-school-id.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DROP INDEX [mtc_admin].pupil.pupil_upn_uindex;
+CREATE UNIQUE NONCLUSTERED INDEX [pupil_upn_uindex] ON [mtc_admin].[pupil]
```
Comment (jon-shipley) [inline admin/data/sql/migrations/20190402121931.do.add-unique-undex-pupil-table-upn-and-school-id.sql]: The  index name must include the school_id component too.

File: admin/data/sql/migrations/20190402121931.do.add-unique-undex-pupil-table-upn-and-school-id.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+(
+  [upn], school_id
```
Comment (jon-shipley) [inline admin/data/sql/migrations/20190402121931.do.add-unique-undex-pupil-table-upn-and-school-id.sql]: Suggest [] quotes are used consistently

File: admin/data/sql/migrations/20190402121931.do.add-unique-undex-pupil-table-upn-and-school-id.sql (hunk 1 above)
Comment (GeorgeChatzigiannis) [reply admin/data/sql/migrations/20190402121931.do.add-unique-undex-pupil-table-upn-and-school-id.sql]: @jon-shipley added.

File: admin/data/sql/migrations/20190402121931.do.add-unique-undex-pupil-table-upn-and-school-id.sql (hunk 2 above)
Comment (GeorgeChatzigiannis) [reply admin/data/sql/migrations/20190402121931.do.add-unique-undex-pupil-table-upn-and-school-id.sql]: @jon-shipley added

File: admin/services/data-access/pupil.data.service.js [hunk 3]
```diff
@@ -152,6 +152,36 @@ pupilDataService.sqlFindOneByUpn = async (upn = '') => {
   return R.head(results)
 }
 
+/**
+ * Find a pupil by upn within school
+ * @param {String} upn
+ * @param {Number} schoolId
+ * @return {Promise<void>}
+ */
+pupilDataService.sqlFindOneByUpnAndSchoolId = async (upn = '', schoolId) => {
```
Comment (GuyHarwood) [inline admin/services/data-access/pupil.data.service.js]: why the empty string default, is that a possible case?

Comment (GeorgeChatzigiannis) [reply admin/services/data-access/pupil.data.service.js]: @GuyHarwood I made a change where lookup should be prevented if the upn that is passed down has already validation errors.
The data service is now modified as well with no default as a value expectation

**Pull request merged**